wrappers/services.go

```diff
@@ -42,46 +44,150 @@ import (
 	"github.com/snapcore/snapd/wrappers/internal"
 )
 
+// ServiceScope represents the scope of services that can be present in a snap, which
+// may be affected by an operation.
+type ServiceScope string
+
+const (
+	// ServiceScopeAll represents both system and user services of a snap.
+	ServiceScopeAll ServiceScope = ""
+	// ServiceScopeSystem represents just the system services of a snap.
+	ServiceScopeSystem ServiceScope = "system"
+	// ServiceScopeUser represents just the user services of a snap.
+	ServiceScopeUser ServiceScope = "user"
+)
+
+func (sc ServiceScope) matches(dscope snap.DaemonScope) bool {
+	switch sc {
+	case ServiceScopeAll:
+		return true
+	case ServiceScopeSystem:
+		return dscope == snap.SystemDaemon
+	case ServiceScopeUser:
+		return dscope == snap.UserDaemon
+	}
+	return false
+}
+
 type Interacter interface {
 	Notify(status string)
 }
 
 // wait this time between TERM and KILL
 var killWait = 5 * time.Second
 
-func stopUserServices(cli *client.Client, inter Interacter, services ...string) error {
+type userServiceClient struct {
+	cli   *client.Client
+	inter Interacter
+}
+
+func usersToUids(users []string) (map[int]string, error) {
+	uids := make(map[int]string)
+	if strutil.ListContains(users, "users") {
```

Comment:
this convention is strange, I would expect to use all for that?

Reply:
we should discuss about this, we probably need to agree on details of the API command line